### PR TITLE
Remove old rust code and reorganise 

### PR DESCRIPTION
**Pull request by msmanoj**
TODOs:
@kamaldodiya 
We need to update the following lines to fix the deployment:
https://gitlab.ebi.ac.uk/kamal/ensembl-client-nginx/-/blob/master/Dockerfile#L12
https://gitlab.ebi.ac.uk/kamal/ensembl-client-nginx/-/blob/master/Dockerfile.prod#L12
to
`COPY dist /usr/staticfiles`

## Type

- [ ] Bug fix
- [ ] New feature
- [ ] Improvement to existing feature
- [x] Maintenance

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1308

## Importance
- It will conflict with most of the PRs

## Description
- Removed the old rust code
- Moved all the contents inside `src/ensembl/*` to root level
- 
## Deployment URL
http://remove-old-rust-code.review.ensembl.org

## Checks
```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run prepare
husky - Git hooks installed
```

```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run prestart
husky - Git hooks installed
```

```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run prod:analyse
99% done plugins webpack-bundle-analyzerWebpack Bundle Analyzer is started at http://127.0.0.1:8888
Use Ctrl+C to close it
The production client build is ready
```

```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run build
The production build is ready
```

```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run  lint
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.4.0

YOUR TYPESCRIPT VERSION: 4.5.2

Please only submit bug reports when using the officially supported version.

=============
FVFY60KAJK7M:ensembl-client manojpandian$ 
```

```
FVFY60KAJK7M:ensembl-client manojpandian$ npm run test
... 
Test Suites: 118 passed, 118 total
Tests:       548 passed, 548 total
Snapshots:   0 total
Time:        179.574 s
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,3 +47,50 @@ We are replacing BioMart with a more scalable and performant solution that allow
 ### Global Search
 
 Similar to our current site-wide search, the Global Search will allow you to search for gene names and symbols and other identifiers.
+
```
Comment (azangru) [inline README.md:50]: Delete line 21 above, which references the old Readme.

At some later point, we'll review the Readme.

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
   "author": "EMBL-EBI",
   "license": "MIT",
   "scripts": {
-    "prepare": "cd ../.. && husky install src/ensembl/.husky",
+    "prepare": "husky install ./.husky",
```
Comment (azangru) [inline package.json:16]: Try just `husky install`. I believe it will use the `.husky` directory by default.

**Review by azangru — APPROVED** [review]
I'm happy with these change. It's up to @kamaldodiya now to coordinate the deployment

**Comment by kamaldodiya** [comment]
@msmanoj  @azangru  The dist folder needs to go from .dockerignore https://github.com/Ensembl/ensembl-client/blob/remove-old-rust-code/.dockerignore#L11

**Comment by azangru** [comment]
> The dist folder needs to go from .dockerignore

That's unfortunate. When testing in Docker locally, it was useful to avoid passing the local `dist` folder to it, to make sure it's not the files built outside of Docker that are picked up by Docker.

**Comment by azangru** [comment]
@kamaldodiya Could you try adding the `.dockerignore` file without the `dist` folder to the same directory that contains `ensembl-client-nginx/Dockerfile`? From a Stack Overflow comment ([link](https://stackoverflow.com/a/57774684/3925302)), it looks like Docker will respect the dockerignore that's in the same directory as the Dockerfile that's used for the build. Then we'll continue using our `.dockerignore` with the ignored dist folder for local testing

**Pull request merged**